cherab/tools/observers/bolometry.py

```diff
@@ -687,6 +695,350 @@ def etendue_single_run(_):
+        pixel_height = self.height / ny
+        # Foil pixels are defined in the foil's local coordinate system
+        foil_bottom_left = Point3D(-self.width / 2, -self.height / 2, 0)
+        pixels = []
+        for x in range(nx):
+            pixel_column = []
+            for y in range(ny):
+                pixel_centre = (foil_bottom_left
+                                + (x + 0.5) * XAXIS * pixel_width
+                                + (y + 0.5) * YAXIS * pixel_height)
+                pixel = BolometerFoil(
+                    detector_id="IRVB pixel ({},{})".format(x + 1, y + 1),
+                    centre_point=pixel_centre, basis_x=XAXIS, dx=pixel_width,
+                    basis_y=YAXIS, dy=pixel_height, slit=self._slit,
+                    units=self.units, accumulate=False, parent=self
+                )
+                pixel_column.append(pixel)
+            pixels.append(pixel_column)
+        return np.asarray(pixels, dtype='object')
+
+    @property
+    def height(self):
+        return self.width * self.pixels[1] / self.pixels[0]
+
+    @property
+    def centre_point(self):
+        return Point3D(0, 0, 0).transform(self.to_root())
+
+    @property
+    def normal_vector(self):
+        return Vector3D(0, 0, 1).transform(self.to_root())
+
+    @property
+    def basis_x(self):
+        return Vector3D(1, 0, 0).transform(self.to_root())
+
+    @property
+    def basis_y(self):
+        return Vector3D(0, 1, 0).transform(self.to_root())
+
+    @property
+    def sightline_vectors(self):
+        return np.asarray(
+            [[pixel.centre_point.vector_to(self._slit.centre_point) for pixel in pixel_column]
+             for pixel_column in self.pixels_as_foils],
+            dtype='object'
+        )
+
+    @property
+    def slit(self):
+        return self._slit
+
+    @property
+    def curvature_radius(self):
+        return self._curvature_radius
+
+    @property
+    def units(self):
+        return self._units
+
+    @units.setter
+    def units(self, units):
+        if units == "Power":
+            pipeline = PowerPipeline2D(accumulate=self.accumulate)
+        elif units == "Radiance":
+            pipeline = RadiancePipeline2D(accumulate=self.accumulate)
+        else:
+            raise ValueError("The units property of BolometerFoil must be one of 'Power' or 'Radiance'.")
+        self._units = units
+        self.pipelines = [pipeline]
+
+    @property
+    def accumulate(self):
+        return self._accumulate
+
+    @accumulate.setter
+    def accumulate(self, value):
+        for pipeline in self.pipelines:
+            pipeline.accumulate = value
+            # Discard any samples from previous accumulate behaviour
+            pipeline.value.clear()
+
+    def as_sightlines(self):
+        """
+        Constructs a SightLine observer for each pixel in this bolometer.
+
+        :return: A 2D array of Sightline objects.
+        """
+
+        if self.units == "Power":
```

Comment:
This string logic should not be repeated, it's crappy to maintain. DRY principle. If you must continue this approach, use constants if these things turn up in multiple places:

```
UNITS_POWER = 'power'
UNITS_RADIANCE = 'radiance'
```

Reply:
I've modified the units logic, using enum's now instead of string literals. It still looks roughly the same from the users' perspective, but internally to the class it's a bit more abstracted. I've tried to write it in such a way that adding or removing possible units values only requires modification to the Enum and some class-level dictionaries: is this the right way to go?